izumi-reflect/izumi-reflect/src/main/scala-2/izumi/reflect/ReflectionUtil.scala

```diff
@@ -139,12 +139,39 @@ private[reflect] object ReflectionUtil {
   }
 
   def kindOf(tpe: Universe#Type): Kind = {
-    Kind(tpe.typeParams.map(t => kindOf(t.typeSignature)))
+    Kind(tpe.typeParams.map { param =>
+      val sig = param.typeSignature
+      val (lo, hi) = sig match {
+        case tb: Universe#TypeBoundsApi => (Some(tb.lo.asInstanceOf[Universe#Type]), Some(tb.hi.asInstanceOf[Universe#Type]))
+        case _ => (None, None)
+      }
+      KindParam(kindOf(sig), lo, hi)
+    })
   }
 
-  final case class Kind(args: List[Kind]) {
-    def format(typeName: String) = s"$typeName${if (args.nonEmpty) args.mkString("[", ", ", "]") else ""}"
+  /** Stores bounds for a type parameter. lo/hi are None for default bounds (Nothing..Any) */
+  final case class KindParam(
```

Comment:
There is no point in distinction between KindParam and Kind.

```suggestion
  final case class Kind(
    args: List[Kind],
    bounds: Option[Universe#TypeBounds]
)
```
@AriajSarkar

Reply:
Thanks, that makes sense. Ill drop KindParam and merge the bounds directly into Kind, following your suggested structure. I’ll update the model and the macro helpers accordingly and push the refactor.